core/src/main/java/org/javarosa/core/model/User.java

```diff
@@ -0,0 +1,290 @@
+        this.wrappedKey = ExtUtil.nullIfEmpty(ExtUtil.readBytes(in));
+    }
+
+    public void writeExternal(DataOutputStream out) throws IOException {
+        ExtUtil.writeString(out, username);
+        ExtUtil.writeString(out, password);
+        ExtUtil.writeNumeric(out, recordId);
+        ExtUtil.writeString(out, ExtUtil.emptyIfNull(uniqueId));
+        ExtUtil.writeBool(out, rememberMe);
+        ExtUtil.writeString(out, ExtUtil.emptyIfNull(syncToken));
+        ExtUtil.write(out, new ExtWrapMap(properties));
+        ExtUtil.writeBytes(out, ExtUtil.emptyIfNull(wrappedKey));
+    }
+
+    public boolean isAdminUser() {
+        return ADMINUSER.equals(getUserType());
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setID(int recordId) {
+
+        this.recordId = recordId;
+    }
+
+    public int getID() {
+        return recordId;
+    }
+
+    public String getUserType() {
+        if (properties.containsKey(KEY_USER_TYPE)) {
+            return properties.get(KEY_USER_TYPE);
+        } else {
+            return null;
+        }
+    }
+
+    public void setUserType(String userType) {
+        properties.put(KEY_USER_TYPE, userType);
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public boolean isRememberMe() {
+        return rememberMe;
+    }
+
+    public void setRememberMe(boolean rememberMe) {
+        this.rememberMe = rememberMe;
+    }
+
+    public void setUuid(String uuid) {
+        this.uniqueId = uuid;
+    }
+
+    public String getUniqueId() {
+        return uniqueId;
+    }
+
+    public Enumeration listProperties() {
+        return this.properties.keys();
+    }
+
+    public void setProperty(String key, String val) {
+        this.properties.put(key, val);
+    }
+
+    public String getProperty(String key) {
+        return this.properties.get(key);
+    }
+
+    public Hashtable<String, String> getProperties() {
+        return this.properties;
+    }
+
+    public String getRestorableType() {
+        return "user";
+    }
+
+    public FormInstance exportData() {
+        FormInstance dm = RestoreUtils.createDataModel(this);
+        RestoreUtils.addData(dm, "name", username);
+        RestoreUtils.addData(dm, "pass", password);
+        RestoreUtils.addData(dm, "uuid", uniqueId);
+        RestoreUtils.addData(dm, "remember", Boolean.valueOf(rememberMe));
+
+        for (Enumeration e = properties.keys(); e.hasMoreElements(); ) {
+            String key = (String)e.nextElement();
+            RestoreUtils.addData(dm, "other/" + key, properties.get(key));
+        }
+
+        return dm;
+    }
+
+    public void templateData(FormInstance dm, TreeReference parentRef) {
```

Comment:
[Minor, skip if you're busy]: Do these export methods still get used? If not can we dump them?